### ISSUE TITLE
Feature request: Include produced event PID

### ISSUE THREAD

**Issue by okynos** [notify-rs/notify#392]
#### System details

- OS/Platform name and version: Linux, Windows and macOS
- Notify version: 4.0.17

- If you're coming from a project that makes use of Notify, what it is, and a link to the downstream issue if there is one: https://github.com/Achiefs/fim/issues/16

Hello all,

First of all thanks for produce Notify crate it is really useful for us.
We want to monitor the produced event process ID to maintain a security list over our files. This is a feature request.
I have done a research into the docs without too much information about that, maybe isn't possible or it isn't provided by the Inotify Daemon.
Anyway thanks for your support.

**Comment by 0xpr03** [comment]
I don't think we can provide that: It's only relevant for the notify system, and thus only for 1 of 5 backends.

If you need that I'd recommend using the inotify interface directly. The only path forward that I could think of, is to find a good amount of backends that do provide something similar (like the windows and fsevent APIs from mac). Thus making it reasonable to add an Option<PID> to the backend agnostic event type.

**Comment by okynos** [comment]
Hi @0xpr03 

Thanks for your reply. If only inotify supports it I will have to use the inotify interface to implement such feature.
I will take a look at it that way. I thought that some more backends support it but sadly don´t.

Again, thanks for your answer and tell me if you have an issue tracking down the v5 feature that you mentioned.
Should I close the issue?

Regards.
